src/parser.rs

```diff
@@ -1268,10 +1268,26 @@ impl Parser {
             ColumnOption::Check(expr)
         } else if self.parse_keyword(Keyword::AUTO_INCREMENT) {
             // Support AUTO_INCREMENT for MySQL
-            ColumnOption::MySQLAutoIncrement
+            let word = Word {
+                value: "AUTO_INCREMENT".to_string(),
+                quote_style: None,
+                keyword: Keyword::AUTO_INCREMENT,
+            };
+            let token = Token::Word(word);
+            let mut vec = vec![];
+            vec.push(token);
+            ColumnOption::DialectSpecific(vec)
```

Comment:
This would be much simpler with `Token::make_keyword()`, I think.

I hoped there was a way to avoid keyword duplication by remembering the position in the token stream before trying to parse. When parsing is successful, we could simply copy the tokens starting from that position into the DialectSpecfic variant. I'm not up to trying to implement that now, so `make_keyword` is fine with me too.